### PR TITLE
Include Kotlin DSL samples for chapter troubleshooting_dependency_resolution

### PR DESCRIPTION
**Pull request by jnizet**
### Gradle Core Team Checklist
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation
- [ ] Recognize contributor in release notes

### CODE REVIEW

File: subprojects/docs/src/samples/userguide/dependencyManagement/troubleshooting/cache/changing/kotlin/build.gradle.kts [hunk 1]
```diff
@@ -0,0 +1,25 @@
+plugins {
+    `java-library`
+}
+
+repositories {
+    mavenCentral()
+    maven {
+        setUrl("https://repo.spring.io/snapshot/")
```
Comment (StefMa) [inline subprojects/docs/src/samples/userguide/dependencyManagement/troubleshooting/cache/changing/kotlin/build.gradle.kts]: there is better overload.
Use `maven(url=....)`

Comment (eskatos) [reply subprojects/docs/src/samples/userguide/dependencyManagement/troubleshooting/cache/changing/kotlin/build.gradle.kts]: :x: this should show:
```kotlin
repositories {
    maven {
        url = uri("...")
    }
}
```

@StefMa, yep, the overload can be handy. But let's stick to the more general and more close to Groovy `maven { url = uri("...") }` for user manual snippets. It also has the side effect of demonstrating how to create and assign a URI in the general case.

File: subprojects/docs/src/samples/userguide/dependencyManagement/troubleshooting/cache/changing/kotlin/build.gradle.kts [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+    implementation("org.springframework:spring-web:5.0.3.BUILD-SNAPSHOT")
+}
+
+// tag::changing-module-cache-control[]
+configurations.all {
+    resolutionStrategy.cacheChangingModulesFor(4, "hours")
+}
+// end::changing-module-cache-control[]
+
+tasks.create<Copy>("copyLibs") {
```
Comment (eskatos) [inline subprojects/docs/src/samples/userguide/dependencyManagement/troubleshooting/cache/changing/kotlin/build.gradle.kts]: :x: task<Copy>("copyLibs") {`

File: subprojects/docs/src/samples/userguide/dependencyManagement/troubleshooting/cache/dynamic/kotlin/build.gradle.kts [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+    implementation("org.springframework:spring-web:5.+")
+}
+
+// tag::dynamic-version-cache-control[]
+configurations.all {
+    resolutionStrategy.cacheDynamicVersionsFor(10, "minutes")
+}
+// end::dynamic-version-cache-control[]
+
+tasks.create<Copy>("copyLibs") {
```
Comment (eskatos) [inline subprojects/docs/src/samples/userguide/dependencyManagement/troubleshooting/cache/dynamic/kotlin/build.gradle.kts]: :x: `task<Copy>("copyLibs") {`

**Comment by jnizet** [comment]
Changes done @eskatos

**Comment by eriwen** [comment]
I fixed the samples tests and merged. Thanks yet again @jnizet!

**Pull request closed without merging**